### DISCUSSION TITLE
Reduce initial window size of WinUI 3 app?

### DISCUSSION THREAD

**Discussion by StevenABrown** [microsoft/microsoft-ui-xaml#9994] [Q&A]
In Visual Studio 2022, I created a project using the WinUI 3 blank project template. The window that appears upon launch of the application takes up about half the screen, and I would like to reduce the size, because I intend to build a small application with few controls. Nowhere in the source files could I find code that specifies the size of the window. How can I do that?

**Comment by castorix** [comment]
You can use this.[AppWindow.MoveAndResize](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.windowing.appwindow.moveandresize?view=windows-app-sdk-1.5) (or just Resize)

**Reply by gautambjain** [reply]
How would I do for any Window object and not just the main window (AppWindow)? ::SetWindowPos is the only solution?

**Reply by gautambjain** [reply]
Sorry didn't read the **this.** you had mentioned. Now I know.

**Comment by mareksm** [comment]
This should do:

`winrt::Microsoft::UI::WindowId windowId = winrt::Microsoft::UI::GetWindowIdFromWindow(hWnd);`
`winrt::Microsoft::UI::Windowing::AppWindow appWindow = winrt::Microsoft::UI::Windowing::AppWindow::GetFromWindowId(windowId);`

`appWindow.Resize(Windows::Graphics::SizeInt32{800, 600});`

**Reply by AlexanderBlackman** [reply]
@mareksm  Where would you put this code?

**Reply by mareksm** [reply]
In a class that implements WindowT, constructor.

**Reply by Balkoth** [reply]
Be careful when setting the size on a display that is scaled other than 100%, because you do need to adjust the values accordingly or your window may be too small or too large.
